livekit-plugins/livekit-plugins-inworld/livekit/plugins/inworld/tts.py

```diff
@@ -0,0 +1,287 @@
+
+        try:
+            await asyncio.gather(*tasks)
+        finally:
+            await utils.aio.cancel_and_wait(*tasks)
+
+    async def _tokenize_input(self, tokenizer_stream: tokenize.SentenceStream) -> None:
+        """Feed input text to the tokenizer stream."""
+        async for text in self._input_ch:
+            if isinstance(text, str):
+                tokenizer_stream.push_text(text)
+            elif isinstance(text, self._FlushSentinel):
+                tokenizer_stream.flush()
+
+        tokenizer_stream.end_input()
+
+    async def _synthesize_sentences(
+        self,
+        tokenizer_stream: tokenize.SentenceStream,
+        output_emitter: tts.AudioEmitter,
+    ) -> None:
+        """Process sentences from tokenizer and synthesize audio."""
+        output_emitter.start_segment(segment_id=shortuuid())
+
+        try:
+            async for sentence in tokenizer_stream:
+                if not sentence.token.strip():
+                    continue
+                await self._synthesize_sentence(sentence, output_emitter)
+            output_emitter.end_segment()
+        except Exception as e:
+            print(f"Error synthesizing segment: {e}")
+            raise
+
+    async def _synthesize_sentence(
+        self, sentence: tokenize.TokenData, output_emitter: tts.AudioEmitter
+    ) -> None:
+        """Synthesize a single sentence to audio."""
+        try:
+            stream = self._create_synthesis_stream(sentence)
+            await self._process_audio_stream(stream, output_emitter)
+        except ValueError as e:
+            if "Chunk too big" in str(e):
+                print(f"Warning: Audio chunk too large, skipping: {e}")
+            else:
+                raise
+        except Exception as e:
+            print(f"Error calling Inworld API: {e}")
+
+    def _create_synthesis_stream(
+        self, sentence: tokenize.TokenData
+    ) -> AsyncGenerator[models.SynthesizeSpeechResponse, None]:
+        """Synthesize a stream for a sentence."""
+        return self._tts._client.tts.synthesizeSpeechStream(
+            input=sentence.token,
+            voice=self._opts.voice,
+            languageCode=self._opts.languageCode,
+            modelId=self._opts.modelId,
+            audioConfig=cast(
+                models.AudioConfig,
+                {
+                    "audioEncoding": AUDIO_ENCODING,
+                    "speakingRate": self._opts.speed,
+                    "sampleRateHertz": self._opts.sampleRateHertz,
+                },
+            ),
+        )
+
+    async def _process_audio_stream(
+        self,
+        stream: AsyncGenerator[models.SynthesizeSpeechResponse, None],
+        output_emitter: tts.AudioEmitter,
+    ) -> None:
+        """Process audio chunks from the synthesized stream."""
+        async for chunk in stream:
+            if chunk and chunk.get("audioContent"):
+                try:
+                    audio_data = base64.b64decode(chunk["audioContent"])
```

Comment:
you don't need to parse this yourself, the output emitter would handle this for you, you can control it by passing in `audio/wav` as mime type

it's not great to duplicate wave parsing code in places.

Reply:
So I'm not getting audio when I try to pass the `chunk["audioContent"]` to the `output_emitter.push()` and initing it as such

```py
        output_emitter.initialize(
            request_id=utils.shortuuid(),
            sample_rate=self._opts.sampleRateHertz,
            num_channels=NUM_CHANNELS,
            mime_type="audio/wav",
            stream=True,
        )
 ```
 
 I'm guessing it's because the content when decoded is including it's mimetype (I've included the first 44 bytes of the decoded chunk for context)
 
 
 ```txt
 RIFF\xec'\x01\x00WAVEfmt \x10\x00\x00\x00\x01\x00\x01\x00\xc0]\x00\x00\x80\xbb\x00\x00\x02\x00\x10\x00data\xc8'\x01\x00
 ```

Reply:
I think this is the only hold up, we synced offline so I'll wait to hear from you regarding this. However I think everything else should be good to go.